sdk/tests/deqp/framework/common/tcuTestCase.js

```diff
@@ -217,8 +219,47 @@ goog.scope(function() {
     * @return {tcuTestCase.DeqpTest}
     */
     tcuTestCase.DeqpTest.prototype.next = function(pattern) {
+        return this._nextHonoringSkipList(pattern);
+    };
+
+    /**
+    * Returns the next test in the hierarchy of tests, honoring the
+    * skip list, and reporting skipped tests.
+    *
+    * @param {?string } pattern Optional pattern to search for
+    * @return {tcuTestCase.DeqpTest}
+    */
+    tcuTestCase.DeqpTest.prototype._nextHonoringSkipList = function(pattern) {
+        var tryAgain = false;
+        var test = null;
+        do {
+            tryAgain = false;
+            test = this._nextIgnoringSkipList(pattern);
+            if (test != null) {
+                // See whether the skip list vetoes the execution of
+                // this test.
+                var fullTestName = test.fullName();
+                var skipDisposition = tcuSkipList.getSkipStatus(fullTestName);
```

Comment:
Thanks for raising this concern. I just tried both putting back in the many transform feedback failure suppressions from #1623 , and commenting out this block of code which tests the skip list at all. There was no visible change in the per-test run time. The skip list is deliberately implemented using JavaScript's built-in property lookup, which should be hash table-based and should be fast.